pkg/k8s/observedstate.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+func HasComputedFieldsUnderObservedState(kind, version string) bool {
+	if version == KCCAPIVersion && !SupportsComputedFieldsUnderStatusInKind(kind) {
```

Comment:
Nit: We need to change KCCAPIVersion to something that includes the word "beta" :-)

Reply:
Agreed that having the version in the name will make it more readable. Updated it in a separate PR to avoid "contaminating" this PR: #1402 .